pkg/management/extensions.go

```diff
@@ -493,10 +494,39 @@ func unknownServiceHandler(director StreamDirector) grpc.StreamHandler {
 }
 
 func decodeAndSetField(reqMsg *dynamic.Message, k, v string) error {
-	fd := reqMsg.FindFieldDescriptorByJSONName(k)
+	fieldPaths := strings.Split(k, ".")
+	if len(fieldPaths) == 0 {
+		return fmt.Errorf("field path %q is invalid", k)
+	}
+	var fd *desc.FieldDescriptor
+	containingMsg := reqMsg
+	for i, fp := range fieldPaths {
+		fd = containingMsg.FindFieldDescriptorByJSONName(fp)
+		if fd == nil {
+			return fmt.Errorf("field path %q is invalid: message %q does not have field %q", k, containingMsg.GetMessageDescriptor().GetName(), fp)
+		}
+		// each field path must be a message, except for the last one
+		if fd.GetType() != dpb.FieldDescriptorProto_TYPE_MESSAGE {
+			if i == len(fieldPaths)-1 {
+				fd = containingMsg.FindFieldDescriptorByJSONName(fp)
+				break
+			}
+			return fmt.Errorf("field path %q is invalid: %q is not a message", k, fd.GetFullyQualifiedName())
+		}
+		msg := fd.GetMessageType()
+		var subMsg *dynamic.Message
+		if fld := containingMsg.GetField(fd); reflect.ValueOf(fld).IsNil() {
+			subMsg = dynamic.NewMessage(msg)
+			containingMsg.SetField(fd, subMsg)
+		} else {
+			subMsg = fld.(*dynamic.Message)
```

Comment:
for consistency with our other impls and to be on the safe side we should probably type check `msg, ok := fld.(*dynamic.Message)`